lib/ecto/changeset.ex

```diff
@@ -362,6 +432,13 @@ defmodule Ecto.Changeset do
     raise ArgumentError, message: "different repos when merging changesets"
   end
 
+  defp merge_identical(object, nil, _thing), do: object
+  defp merge_identical(nil, object, _thing), do: object
+  defp merge_identical(object, object, _thing), do: object
+  defp merge_identical(_, _, thing) do
+    raise ArgumentError, message: "different #{thing} when merging changesets"
```

Comment:
No need for `message: `. Can we also say the values we got in the error message?